api/METRICS.md

```diff
@@ -35,72 +45,260 @@ acc_ratio = accuracy_score_ratio(y_true, y_pred, sensitive_features=sf, **other_
+
+  * `equalized_odds_ratio` <br>
+    = min(`true_positive_rate_ratio`, `false_positive_rate_ratio`)
+
+### fairlearn.postprocessing
+
+**Status quo.** Our postprocessing algorithms use the following calling conventions:
+
+* `constraints` is a string that represents fairness constraints as follows:
+
+  * `"demographic_parity"`: for all _a_, <br>
+    _selection_rate_(_a_) = _selection_rate_(\*).
+
+  * `"equalized_odds"`: for all _a_,<br>
+    _true_positive_rate_(_a_) = _true_positive_rate_(\*), <br>
+    _false_positive_rate_(_a_) = _false_positive_rate_(\*).
+
+* `objective` is always _accuracy_score_(\*).
+
+**Proposal.** The following proposal is an extension of the status quo; no breaking changes are introduced:
+
+* `constraints`: in addition to `"demographic_parity"` and `"equalized_odds"`, also allow:
+
+  * `"<metric>_parity"`: for all _a_,<br>
+    _metric_(_a_) = _metric_(\*).
+
+* `objective` is a string of the form:
+
+  * `"<metric>"`: <br>
+    goal is then to maximize _metric_(\*) subject to constraints
+
+### fairlearn.reductions [TODO]
+
+We support two algorithms: `ExponentiatedGradient` and `GridSearch`. They both represent `constraints` and `objective` via objects of type `Moment`.
+
+**Status quo:**
+
+* In both cases, `objective` is automatically inferred from the provided `constraints`.
+
+* `ExponentiatedGradient(estimator, constraints, eps=`&epsilon;`)` considers constraints that are specified jointly by the provided `Moment` object and the numeric value &epsilon; as follows:
+
+  * `DemographicParity()`: for all _a_, <br>
+    |_selection_rate_(_a_) - _selection_rate_(\*)| &le; &epsilon;.
+
+  * `DemographicParity(ratio=`_r_`)`: for all _a_, <br>
+    _r_ &sdot; _selection_rate_(_a_) - _selection_rate_(\*) &le; &epsilon;, <br>
+    _r_ &sdot; _selection_rate_(\*) - _selection_rate_(_a_) &le; &epsilon;.
+
+  * `TruePositiveRateDifference()`: for all _a_, <br>
+    |_true_positive_rate_(_a_) - _true_positive_rate_(\*)| &le; &epsilon;.
+
+  * `TruePositiveRateDifference(ratio=`_r_`)`: analogous
+
+  * `EqualizedOdds()`: for all _a_, <br>
+    |_true_positive_rate_(_a_) - _true_positive_rate_(\*)| &le; &epsilon;, <br>
+    |_false_positive_rate_(_a_) - _false_positive_rate_(\*)| &le; &epsilon;.
+
+  * `EqualizedOdds(ratio=`_r_`)`: analogous
+
+  * `ErrorRateRatio()`: for all _a_, <br>
+    |_error_rate_(_a_) - _error_rate_(\*)| &le; &epsilon;.
+
+  * `ErrorRateRatio(ratio=`_r_`)`: analogous
+
+  * all of the above constraints are descendants of `ConditionalSelectionRate`
+
+  * the `objective` for all of the above constraints is `ErrorRate()`
+
+* `GridSearch(estimator, constraints)` considers constraints represented by the provided `Moment` object. The behavior of the `GridSearch` algorithm does not depend on the value of the right-hand side of the constraints, so it is not provided to the constructor. In addition to the `Moment` objects above, `GridSearch` also supports the following:

```

Comment:
The 'right-hand side of the constraints' being \epsilon here? i.e. GridSearch produces all your options, and the user is able to determine which best satisfies their requirements?

Reply:
For BGL, the right-hand side is not \epsilon, but \zeta. GridSearch by default optimizes the sum of objective and max(gamma) (which _almost_ corresponds to a shifted constraint violation). So things are a bit confusing there. With the moment objects modified as outlined in this proposal (i.e., they'll include the right-hand-side of the constraint), we will be able to make GridSearch behavior more systematic.